### ISSUE TITLE
Minor: Upvar specs should show ... and the symbol in the block editor.

### ISSUE THREAD

**Issue by cycomachead** [jmoenig/Snap--Build-Your-Own-Blocks#868]
It would be nice if the prototype for the labels for editing blocks that have upvars could shoud show you the input type in addition to the "...".

**Comment by jmoenig** [comment]
I don't understand. Upvars do show an up-arrow indicating them as ... upvars.

![upvar](https://cloud.githubusercontent.com/assets/2596285/8668019/e06f686c-2a06-11e5-9fe1-cd0e25f27697.png)

**Comment by cycomachead** [comment]
I think I had a major brain and meant to say "variadic inputs".

...too much multitasking right now...

**Comment by jmoenig** [comment]
Ah, I get it, thanks :-)

Of course, at one point concatenating multiple cryptic symbols produces hieroglyphs ...

**Comment by cycomachead** [comment]
> Of course, at one point concatenating multiple cryptic symbols produces hieroglyphs ...

Yeah, I'm not sure how some of the combinations would look, there would need to be some spacing like "nums... #" but I've been looking at more and more script-pics recently with variadic inputs, and this would be a handy message.

**Comment by jmoenig** [comment]
I see, it's more for reading (and correcting, I guess). How about type-first:

    nums #...
    tests ?...
    funs λ...

(nobody ever uses anything else than nums or any, right?)

**Comment by cycomachead** [comment]
That actually doesn't look bad!

We do have plenty of variadic list inputs! I should show you something we just gave to students for unit testing. It's really quite nice. :)

Nested lists would almost make a box, which is a 2D object, which is usually what nested lists are, so it almost looks like what it represents!

![test def](https://cloud.githubusercontent.com/assets/1505907/8668307/93dbfe22-29be-11e5-9256-8f9bc6cf48d3.png)
![test](https://cloud.githubusercontent.com/assets/1505907/8668308/93dd5b6e-29be-11e5-8f9b-fa781ca683ce.png)


> (nobody ever uses anything else than nums or any, right?)

Once variadic C-shapes look better I might use those. (I was actually going to use variadic predicate reporters for a more advanced version of defining custom tests that weren't just equality.)

**Comment by cycomachead** [comment]
(To be clear, this was showing a failing case. The `+` operator does work correctly. :grinning: )

**Comment by jmoenig** [comment]
Nice! I guess we should do it, right?

**Comment by cycomachead** [comment]
I think it'd be useful. "_

**Comment by jmoenig** [comment]
is this something urgent? Should we try to get it into the next release (before the Scratch conference)?

**Comment by cycomachead** [comment]
> is this something urgent? Should we try to get it into the next release (before the Scratch conference)?

This one is definitely not urgent, just something helpful.  :)

**Comment by bb010g** [comment]
[**@jmoenig**](https://github.com/jmoenig) commented on [Jul 14, 2015, 12:22 AM PDT](https://github.com/jmoenig/Snap--Build-Your-Own-Blocks/issues/868#issuecomment-121151007 "2015-07-14T07:22:06Z - Replied by Github Comment Enhancer"):
> I see, it's more for reading (and correcting, I guess). How about type-first:
> 
>     nums #...
>     tests ?...
>     funs λ...
> 
> (nobody ever uses anything else than nums or any, right?)

Personal note: I think `nums : #...` would be nicer to heighten the distinction between var name and type. Also, a tinge of preparation for any formal type theory later on in life.
